lib/ansible/modules/windows/win_inet_proxy.py

```diff
@@ -0,0 +1,154 @@
+      hostname and/or port for the protocol specified.
+    - Valid protocols are C(http), C(https), C(ftp), and C(socks).
+    - Omit, set to null or an empty string to remove the proxy settings.
+  connection:
+    description:
+    type: str
+notes:
+- This is not the same as the proxy settings set in WinHTTP through the
+  C(netsh) command. Use the M(win_http_proxy) module ot manage that instead.
+- These settings are by default set per user and not system wide. A registry
+  property must be set independently from this module if you wish to apply the
+  proxy for all users. See examples for more detail.
+- If per user proxy settings are desired, use I(become) to become any local
+  user on the host. No password is needed to be set for this to work.
+- If the proxy requires authentication, set the credentials using the
+  M(win_credential) module. This requires I(become) to be used so the
+  credential store can be accessed.
+seealso:
+- module: win_http_proxy
+- module: win_credential
+author:
+- Jordan Borean (@jborean93)
+'''
+
+EXAMPLES = r'''
+# This should be set before running the win_inet_proxy module
+- name: Configure IE proxy settings to apply to all users
+  win_regedit:
+    path: HKLM:\SOFTWARE\Policies\Microsoft\Windows\CurrentVersion\Internet Settings
+    name: ProxySettingsPerUser
+    data: 0
+    type: dword
+    state: present
+
+# This should be set before running the win_inet_proxy module
+- name: Configure IE proxy settings to apply per user
+  win_regedit:
+    path: HKLM:\SOFTWARE\Policies\Microsoft\Windows\CurrentVersion\Internet Settings
+    name: ProxySettingsPerUser
+    data: 1
+    type: dword
+    state: present
+
+- name: Configure IE proxy to use auto detected settings without an explicit proxy
+  win_inet_proxy:
+    auto_detect: yes
+
+- name: Configure IE proxy to use auto detected settings with a configuration script
+  win_inet_proxy:
+    auto_detect: yes
+    auto_config_url: http://proxy.ansible.com/proxy.pac
+
+- name: Configure IE to use explicit proxy host
+  win_inet_proxy:
+    auto_detect: yes
+    proxy: ansible.proxy
+
+- name: Configure IE to use explicit proxy host with port and without auto detection
+    auto_detect: no
+    proxy: ansible.proxy:8080
+
+- name: Configure IE to use a specific proxy per protocol
+  win_inet_proxy:
+    proxy:
+      http: ansible.proxy:8080
+      https: ansible.proxy:8443
+
+- name: Configure IE to use a specific proxy per protocol using a string
+  win_inet_proxy:
+    proxy: http=ansible.proxy:8080;https=ansible.proxy:8443
+
+- name: Set a proxy with a bypass list
+  win_inet_proxy:
+    proxy: ansible.proxy
+    bypass:
+    - server1
+    - server2
+    - <-loopback>
+    - <local>
+
+- name: Remove any explicit proxies that are set
+  win_inet_proxy:
+    proxy: ''
+    bypass: ''
+
+# This should be done after setting the IE proxy with win_inet_proxy
+- name: Import IE proxy configuration to WinHTTP
+  win_http_proxy:
+    source: ie
+'''
```

Comment:
Suggest adding a combined example that also uses win_credential.

Reply:
Good idea, I forgot to actually test it out so I'm doing that right now.